### ISSUE TITLE
Provide a way to disable rules via flags

### ISSUE THREAD

**Issue by mglaman** [mglaman/phpstan-drupal#385]
Kind of a dupe of https://github.com/mglaman/phpstan-drupal/issues/295 but is also a different approach.

With the extension installer there is no way to split the configuration of "basic setup" and then "additional rules" as far as I can tell. What can be done is adding flags such as:

* disablePluginManagerInspection
* ignoreInternalClassUsage

And so on. Basically opting out of any rule which does more than deprecations detection.

**Comment by mxr576** [comment]
How about just relying on the built-in solution by PHPStan?
https://phpstan.org/user-guide/ignoring-errors

Or anybody who really need go "rogue" can create their own ruleset without depending on the those that comes from this library, can't they?

**Comment by mglaman** [comment]
You're correct, they could. If I moved the rules to a `rules.neon` and updated the `composer.json` to bundle both as part of extension-installer. Then they could choose _not_ to use that package to exclude the rules.

**Comment by mxr576** [comment]
Leaving this here as a reference, IIRC we talked about this on Twitter; adding something similar as a solution here:

https://github.com/phpstan/phpstan-strict-rules/commit/f120c3b73e01e9330c8a0cfc408f0c3f94ac6a89

**Comment by dpi** [comment]
Can we use https://phpstan.org/error-identifiers ?

**Comment by mglaman** [comment]
I have https://github.com/mglaman/phpstan-drupal/issues/772
